File: app/src/main/java/com/doHabit/models/FirebaseProgramView.java
package com.dohabit.models;

import com.google.firebase.database.DataSnapshot;

public class FirebaseProgramView implements ProgramViewProvider {
    private final String title;
    private final String percent;
    private final String description;
    private final String imageLink;
    private final DataSnapshot snapshot;

    public FirebaseProgramView(final DataSnapshot snapshot) {
        this.snapshot = snapshot;
        title = snapshot.child("name").getValue(String.class);
        percent = String.format("%s SUCCESS", snapshot.child("success").getValue(String.class));
        description = snapshot.child("description").getValue(String.class);
        imageLink = String.format("http://habbitsapp.esy.es/img_progs/%s.jpg",
                snapshot.child("image").getValue(String.class));
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public String getPercent() {
        return percent;
    }

    @Override
    public String getImageLink() {
        return imageLink;
    }

    @Override
    public DataSnapshot getSnapshot() {
        return snapshot;
    }
}

File: app/src/main/java/com/doapps/habits/models/FirebaseProgramView.kt
package com.doapps.habits.models

import com.google.firebase.database.DataSnapshot

class FirebaseProgramView(override val snapshot: DataSnapshot) : IProgramViewProvider {
  override val title: String? =
      snapshot.child("name").getValue(String::class.java)

  override val percent: String? =
      "${snapshot.child("success").getValue(String::class.java)} SUCCESS"

  override val description: String? =
      snapshot.child("description").getValue(String::class.java)

  override val imageLink: String? = "http://habit.esy.es/img_progs/" +
      "${snapshot.child("image").getValue(String::class.java)}.jpg"
}